README.md

```diff
@@ -3,94 +3,97 @@
-  chapters.
+There are usually 2 approaches to working on a chapter:
+
+* Focusing on the end-user functionality and do only as much as technically required to tackle **just** the current chapter.
+* Focusing on trying to implement a robust design with the idea of reusing the code in the future chapters.
 
-Either approach works just fine for the purpose of StockPerson and it's mostly a matter of personal
-preference.
+Either approach works just fine for the purpose of StockPerson and it's mostly a matter of personal preference.
 
-# Chapters
+## Chapters
 
 ### 📗 Chapter 1.0
 
-1. Read a CSV file into memory containing sales invoice lines.
-2. Pretty-print the invoices.
+1. Read a CSV file that contains sales invoice lines.
+2. Pretty-print the invoices in a document style.
+
+> [!NOTE]
+> If you don't what exactly "pretty-print" entails, you take a look at an example [here.](./assets/prettyprint_example.txt)
 
 ### 📗 Chapter 2.0
 
-1. Read a CSV file into memory containing sales invoice lines.
+1. Read a CSV file that contains sales invoice lines.
 2. Find the total sales.
-3. Find the most expensive invoice and pretty print it.
+3. Find the most expensive invoice and pretty print it in a document style.
 4. Find the most expensive product and print its code.
-5. Calculate the average price of each product, and print the the list of prices in descending
-   order.
-6. Find the total sales per customer and print a list in descending order in the form "customer,
-   total".
+5. Calculate the average price of each product, and print the the list of prices in descending order.
+6. Find the total sales per customer and print a list in descending order in the form `customer, total`.
 7. Find the customer with the largest total sales.
-8. Find the 3 customers with the least total sales and print a list in descending format in the form
-   "customer, total".
+8. Find the 3 customers with the least total sales and print a list in descending format in the form `customer, total`.
 9. Find the date with the largest total sales amount.
 
 ### 📗 Chapter 3.0
 
-1. Read a CSV file into memory containing product inventory.
-2. Read a CSV file into memory which contain sales invoice lines.
+> [!IMPORTANT]
+> It is important to process the invoices in the order they appear in the CSV.
+
+1. Read a CSV file that contains product inventory.
+2. Read a CSV file that contains sales invoice lines.
 3. Check if the product referenced on each line has enough inventory.
-   1. If no, do not import the invoice to which the line belongs.
-   2. If yes, import the invoice line and update the inventory of the product accordingly.
-4. It is important to process the invoices in the order they appear in the CSV.
-5. Print a list of invoices listing the problematic line(s) for each invoice.
+	1. If no, do not import the invoice to which the line belongs.
+	2. If yes, import the invoice line and update the inventory of the product accordingly.
```

Comment:
Can you use 2 or 4 spaces instead of TABs here?

Reply:
I do kind of prefer tabs, so I changed everything to tabs 😅
I'll change everything back 👍

Reply:
I think that's everything.